lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -843,9 +849,20 @@ class DxilFinalizeModule : public ModulePass {
 
       if (!IsLib) {
         // Set used masks for signature elements
-        MarkUsedSignatureElements(DM.GetEntryFunction(), DM);
+        MarkUsedSignatureElements(DM.GetEntryFunction(), DM, IsLib);
         if (DM.GetShaderModel()->IsHS())
```

Comment:
Here is how you would just pass the entry props, see next comment for hull shader function.
```suggestion
        Function *F = DM.GetEntryFunction();
        // Assume entry props are available for entry function.
        const DxilEntryProps &entryProps = DM.GetDxilEntryProps(F);
        MarkUsedSignatureElements(F, entryProps);
        if (props.IsHS() && props.ShaderProps.HS.patchConstantFunc)
```
Note: don't need to pass DxilModule anymore.